src/pkg/p2p/preheat/dao/instance_test.go

```diff
@@ -0,0 +1,116 @@
+package dao
```

Comment:
Provider instance DAO and manager seem to be @chlins' work.  @chlins, could you please take a detailed review of such parts?